File: Data Structures/Queues/QueueList.py

#   class node to create a node for the queue linked list
class Node : 

    def __init__(self, val) : 
        self.val = val
        self.next = None


class Queue : 

    #   contructor of the queue class
    def __init__(self) : 
        self.front = None
        self.rear = None


    #   method to insert an element into the queue
    def insert(self, val) : 
        new_node = Node(val)

        #   if the front and rear pointer are null by default
        if self.rear is None and self.front is None : 
            self.rear = new_node
            self.front = new_node

        #   when there is already one more nodes or elements in the queue
        else : 
            self.rear.next = new_node
            self.rear = new_node


    def traverse(self) : 
        ptr = self.front
        if ptr is None : 
            print('Queue is empty')
            return
        else : 
            while ptr is not None : 
                print(ptr.val, end = " ")
                ptr = ptr.next       


    def delete(self) : 
        ptr = self.front
        if ptr is None : 
            print("queue is empty")
            return
        else : 
            print(ptr.val)
            self.front = ptr.next
            del ptr


if __name__ == '__main__' : 
    queue = Queue()
    queue.insert(1)
    queue.insert(2)
    queue.insert(3)
    queue.delete()
    queue.delete()
    queue.delete()
    queue.delete()
    queue.traverse()

File: Data Structures/Queues/QueueList.java

class Node {
    protected int val;
    protected Node next;

    public Node () {
        this.val = 0;
        this.next = null;
    }

    public Node(int sk) {
        this.val = sk;
        this.next = null;
    }
}

class Queue{
    private Node front, rear;

    public Queue() {
        this.front = null;
        this.rear = null;
    }

    public void insert(int sk) {
        Node newnode = new Node(sk);
        if (this.rear == null) {
            this.front = this.rear = newnode;
        }
        else{
            this.rear.next = newnode;
            this.rear = newnode;
        }
    }

    public void traverse() {
        if(this.front == null) {System.out.println("Queue is empty");}
        Node ptr = this.front;
        while (ptr!=null) {
            System.out.print(ptr.val + " ");
            ptr = ptr.next;
        }
    }

    public int delete() {
        if(this.front == null) {
            System.out.println("Queue is empty");
            return -1;
        }
        Node ptr = this.front;
        this.front = ptr.next;
        int temp = ptr.val;
        ptr = null;
        return temp;
    }
}


public class QueueList {
    public static void main(String[] args) {
        Queue queue = new Queue();
        queue.insert(1);
    }
}
